internal/grpcserver/storage/redis.go

```diff
@@ -0,0 +1,167 @@
+package storage
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/astanishevskyi/grpc-server/internal/grpcserver/models"
+	"github.com/go-redis/redis/v8"
+	"log"
+	"os"
+	"strconv"
+	"sync"
+)
+
+type RedisStorage struct {
+	RedisStorage *redis.Client
+	lastID       uint32
+	mu           sync.Mutex
+}
+
+func NewRedisStorage() *RedisStorage {
+	addr := os.Getenv("REDIS_ADDR")
+	password := os.Getenv("REDIS_PASS")
+	db, err := strconv.Atoi(os.Getenv("REDIS_DB"))
+	if err != nil {
+		log.Fatal(err)
+	}
+	rdb := redis.NewClient(&redis.Options{
+		Addr:     addr,
+		Password: password,
+		DB:       db,
+	})
+	storage := &RedisStorage{RedisStorage: rdb}
+	lastID := storage.getLastID()
+	storage.lastID = lastID
+	return storage
+}
+
+func (r *RedisStorage) getLastID() uint32 {
+	var cursor uint64
+	var lastID uint32
+	var err error
+	var result []string
+	for {
+		result, cursor, err = r.RedisStorage.Scan(context.Background(), cursor, "*", 10).Result()
+		if err != nil {
+			log.Fatal(err)
+		}
+		for _, i := range result {
+			id, err := strconv.Atoi(i)
+			if err != nil {
+				log.Fatal(err)
+			}
+			if lastID < uint32(id) {
+				lastID = uint32(id)
+			}
+		}
+		if cursor == 0 {
+			break
+		}
+	}
+	return lastID
+}
+
+func (r *RedisStorage) GetAll() ([]models.User, error) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	var cursor uint64
+	var result []models.User
+	for {
+		res, cursor, err := r.RedisStorage.Scan(context.Background(), cursor, "*", 10).Result()
```

Comment:
maybe you'd like to pass `context` as an argument to the function in order to close the connection if smth outside of the function happened